### PR TITLE
[BUGFIX] Make QueryResultToArrayDynamicReturnTypeExtension work with CompoundType

### LINKED ISSUE #103: [Bug] Error when calling `toArray()` on repository method in controller
Since the latest version v1.8.0 we get this error when running PHPStan:

```
Internal error: Internal error: Repository class "Vendor\Package\Controller\MyController" must implement "TYPO3\CMS\Extbase\Persistence\RepositoryInterface" in file 
     /var/www/html/packages/package/Classes/MyController.php

[...]

#0 /var/www/html/vendor/saschaegerer/phpstan-typo3/src/Type/QueryResultToArrayDynamicReturnTypeExtension.php(52):                                                                                                   
     SaschaEgerer\PhpstanTypo3\Type\QueryResultToArrayDynamicReturnTypeExtension->translateRepositoryNameToModelName('Vendor\\MyPackage...')
```

Related line:
```php
$items = $this->myRepository->findAll()->toArray();
```

### PR DESCRIPTION
**Pull request by sascha-egerer**
Resolves #103

### CODE REVIEW

File: src/Type/QueryResultToArrayDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -43,6 +44,8 @@ public function getTypeFromMethodCall(
 		$resultType = $scope->getType($methodCall->var);
 		if ($resultType instanceof GenericObjectType) {
 			$modelType = $resultType->getTypes();
+		} elseif($resultType instanceof CompoundType && $resultType->getTypes()[0] instanceof GenericObjectType) {
```
Comment (sascha-egerer) [inline src/Type/QueryResultToArrayDynamicReturnTypeExtension.php:47]: This is IMO not a really nice solution but works for now. If someone has time to investigate how to make it more stable feel free to contribute.

Comment (sascha-egerer) [reply src/Type/QueryResultToArrayDynamicReturnTypeExtension.php]: I've adjusted this bit.

**Comment by sabbelasichon** [comment]
@sascha-egerer I would create a failing test first. I have done that locally, but your fix doesn´t solve the error for me.

**Comment by dahaupt** [comment]
We still have the error with the patch as well. The `$resultType` from `getGenericTypes()` is `null`.

**Comment by sascha-egerer** [comment]
> @sascha-egerer I would create a failing test first. I have done that locally, but your fix doesn´t solve the error for me.

Could you please provide the test you did?

**Comment by sabbelasichon** [comment]
I have added the failing test on top of your commits.

**Comment by sascha-egerer** [comment]
@dahaupt Could you please give it another try?

**Review by sabbelasichon — APPROVED** [review]

**Comment by dahaupt** [comment]
@sascha-egerer Please see comment in related issue.

**Pull request merged**